.github/workflows/twisted_trunk.yml

```diff
@@ -55,11 +65,24 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - name: Patch dependencies
-        run: .ci/patch_for_twisted_trunk.sh
+        # Note: The script below runs poetry against a virtualenv in /src/.venv/
+        #       ...but the sytest-synapse container expects it to be in /venv/
+        #       So we temporarily symlink it, run poetry, then remove the link.
+        #       Tidying up is important because `/src` is a mounted volume.
+        run: |
+          ln -s -T /venv /src/.venv
+          poetry remove twisted
+          poetry add --extras tls git+https://github.com/twisted/twisted.git#trunk
+          poetry install --no-interaction --extras "all test"
+          rm /src/.venv
```

Comment:
```suggestion
        # Note: The poetry commands want to create a virtualenv in /src/.venv/,
        #       but the sytest-synapse container expects it to be in /venv/,
        #       so we symlink it before running poetry so that poetry actually
        #       ends up installing to `/venv`.
        run: |
          ln -s -T /venv /src/.venv
          poetry remove twisted
          poetry add --extras tls git+https://github.com/twisted/twisted.git#trunk
          poetry install --no-interaction --extras "all test"
```